### PR TITLE
use the current language version for spawnHybridCode

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/test/issues/1363

### CODE REVIEW

**Comment by jakemac53** [comment]
Probably will have to update some tests etc but wanted to see what happens on CI

File: pkgs/test_core/lib/src/runner/spawn_hybrid.dart [hunk 1]
```diff
@@ -110,11 +111,14 @@ String _normalizeUrl(String url, Suite suite) {
 ///
 /// If there is a language version comment in the file, that is returned.
 ///
+/// If the uri is a `data` uri, a comment representing the language version of
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/spawn_hybrid.dart:114]: [nit] 
```suggestion
/// If the URI has a `data` scheme, a comment representing the language version of
```

Capitalize URI when used in comments.

File: pkgs/test_core/lib/src/runner/spawn_hybrid.dart [hunk 2]
```diff
@@ -110,11 +111,14 @@ String _normalizeUrl(String url, Suite suite) {
...
+/// the current package is returned.
+///
 /// Otherwise a comment representing the default version from the
 /// [currentPackageConfig] is returned.
 ///
-/// If no default language version is known (data: uri for instance), then
-/// an empty string is returned.
+/// If no default language version is known (the uri scheme is not recognized
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/spawn_hybrid.dart:120]: [nit]

```suggestion
/// If no default language version is known (the URI scheme is not recognized
```

File: pkgs/test_core/lib/src/runner/spawn_hybrid.dart [hunk 3]
```diff
@@ -127,15 +131,19 @@ Future<String> _languageVersionCommentFor(String url) async {
...
-      parsedUri.scheme == 'file' ||
-      parsedUri.scheme == 'data') {
+  if (parsedUri.scheme.isEmpty || parsedUri.scheme == 'file') {
     var packageConfig = await currentPackageConfig;
     var package = packageConfig.packageOf(parsedUri);
     var version = package?.languageVersion;
     if (version != null) return '// @dart=$version';
   }
 
+  // Returns the root package language version for `data` uris. These are
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/spawn_hybrid.dart:141]: [nit]
```suggestion
  // Returns the root package language version for `data` URIs. These are
```

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
cc @natebosch PTAL I added an integration test directory and an opted out package to verify the behavior

**Review by natebosch — APPROVED** [review]

**Pull request merged**